internal/internal_retry_test.go

```diff
@@ -0,0 +1,46 @@
+package internal
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	s "go.uber.org/cadence/.gen/go/shared"
+)
+
+func TestErrRetries(t *testing.T) {
+	t.Run("retryable", func(t *testing.T) {
+		for _, err := range []error{
+			&s.CurrentBranchChangedError{},      // unsure, but second attempt may get the latest branch?
```

Comment:
FYI, `CurrentBranchChangedError`, `RetryTaskV2Error`, `InternalDataInconsistencyError` and `RemoteSyncMatchedError` are all internal errors and should never even appear for client. In protobuf, they were even [moved out](https://github.com/uber/cadence/blob/master/proto/internal/uber/cadence/shared/v1/error.proto) of public IDLs.  Probably does not matter for this PR though.